test/conformance/ingress/run.go

```diff
@@ -17,43 +17,59 @@ limitations under the License.
 package ingress
 
 import (
+	"strings"
 	"testing"
 
 	"knative.dev/networking/test"
 )
 
+var stableTests = map[string]func(t *testing.T){
+	"basics":                       TestBasics,
+	"basics/http2":                 TestBasicsHTTP2,
+	"grpc":                         TestGRPC,
+	"grpc/split":                   TestGRPCSplit,
+	"headers/pre-split":            TestPreSplitSetHeaders,
+	"headers/post-split":           TestPostSplitSetHeaders,
+	"hosts/multiple":               TestMultipleHosts,
+	"dispatch/path":                TestPath,
+	"dispatch/percentage":          TestPercentage,
+	"dispatch/path_and_percentage": TestPathAndPercentageSplit,
+	"timeout":                      TestTimeout,
+	"tls":                          TestIngressTLS,
+	"update":                       TestUpdate,
+	"visibility":                   TestVisibility,
+	"visibility/split":             TestVisibilitySplit,
+	"visibility/path":              TestVisibilityPath,
+	"ingressclass":                 TestIngressClass,
+	"websocket":                    TestWebsocket,
+	"websocket/split":              TestWebsocketSplit,
+}
+
+var betaTests = map[string]func(t *testing.T){
+	// Add your conformance test for beta features
+	"headers/probe": TestProbeHeaders,
+}
+
+var alphaTests = map[string]func(t *testing.T){
+	// Add your conformance test for alpha features
+	"headers/tags": TestTagHeaders,
+	"host-rewrite": TestRewriteHost,
+}
+
 // RunConformance will run ingress conformance tests
 //
 // Depending on the options it may test alpha and beta features
 func RunConformance(t *testing.T) {
-	t.Run("basics", TestBasics)
-	t.Run("basics/http2", TestBasicsHTTP2)
-
-	t.Run("grpc", TestGRPC)
-	t.Run("grpc/split", TestGRPCSplit)
-
-	t.Run("headers/pre-split", TestPreSplitSetHeaders)
-	t.Run("headers/post-split", TestPostSplitSetHeaders)
-
-	t.Run("hosts/multiple", TestMultipleHosts)
 
-	t.Run("dispatch/path", TestPath)
-	t.Run("dispatch/percentage", TestPercentage)
-	t.Run("dispatch/path_and_percentage", TestPathAndPercentageSplit)
-
-	t.Run("timeout", TestTimeout)
-
-	t.Run("tls", TestIngressTLS)
-	t.Run("update", TestUpdate)
-
-	t.Run("visibility", TestVisibility)
-	t.Run("visibility/split", TestVisibilitySplit)
-	t.Run("visibility/path", TestVisibilityPath)
-
-	t.Run("ingressclass", TestIngressClass)
+	for name, test := range stableTests {
+		t.Run(name, test)
+	}
 
-	t.Run("websocket", TestWebsocket)
-	t.Run("websocket/split", TestWebsocketSplit)
+	skipTests := strings.Split(test.ServingFlags.SkipTests, ",")
+	for _, name := range skipTests {
+		delete(alphaTests, name)
+		delete(betaTests, name)
```

Comment:
It would be good to create a set of skipped test names and invoke `t.Skip()` so you get some output that a test was skipped

Reply:
That is nice idea but `t.Skip()` calls against "root" test :cry:. So It outputs `SKIP: TestIngressConformance` even though each conformance test outputs `PASS` like this:

```
--- SKIP: TestIngressConformance (0.00s)
    --- PASS: TestIngressConformance/headers/tags (8.02s)
        --- PASS: TestIngressConformance/headers/tags/matching_tag_header (1.29s)
        --- PASS: TestIngressConformance/headers/tags/no_tag_header (0.32s)
        --- PASS: TestIngressConformance/headers/tags/empty_tag_header (0.12s)
        --- PASS: TestIngressConformance/headers/tags/non-matching_tag_header (0.11s)
```

It is strange... So I think we should avoid to call `t.Skip` in run.go

Reply:
You could artificially call skip on the subtest - ie.

```go
if test.ServingFlags.EnableAlphaFeatures {
  for name, test := range alphaTests {
   if skipTest(name) {
    t.Run(name, func(t *t.testing.T) { t.Skip() }
    continue
  }
   t.Run(name, test)
  }
}
```

Reply:
Every test calls `test.Setup(t)`, doesn't it? How about putting it in there?

Reply:
Thank you. Updated. The current result shows like this:

```
=== RUN   TestIngressConformance/host-rewrite
    TestIngressConformance/host-rewrite: run.go:107: Skipping the test in skip-test flag

   ... 

--- PASS: TestIngressConformance (0.00s)
    --- SKIP: TestIngressConformance/host-rewrite (0.00s)
    --- PASS: TestIngressConformance/headers/tags (15.67s)
        --- PASS: TestIngressConformance/headers/tags/matching_tag_header (0.61s)
        --- PASS: TestIngressConformance/headers/tags/no_tag_header (0.11s)
        --- PASS: TestIngressConformance/headers/tags/empty_tag_header (0.10s)
        --- PASS: TestIngressConformance/headers/tags/non-matching_tag_header (0.21s)
```

Reply:
@coryrc Thank you. But `Setup` is a function to setup test client. I feel it is a little bit weird that we put the `Skip` logic there. Let's go on Dave's suggestion.